plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -202,15 +217,26 @@ export class EventsProcessor {
             })
         }
 
-        const groupIdentifiers = this.getGroupIdentifiers(properties)
-        const groupsColumns = await this.db.getGroupsColumns(teamId, groupIdentifiers)
+        let groupsColumns: Record<string, string | ClickHouseTimestamp> = {}
+        let eventPersonProperties: string = '{}'
+        if (processPerson) {
+            const groupIdentifiers = this.getGroupIdentifiers(properties)
+            groupsColumns = await this.db.getGroupsColumns(teamId, groupIdentifiers)
+            eventPersonProperties = JSON.stringify({
+                ...person.properties,
+                // For consistency, we'd like events to contain the properties that they set, even if those were changed
+                // before the event is ingested.
+                ...(properties.$set || {}),
+            })
+        } else {
+            // TODO: Move this into `normalizeEventStep` where it belongs, but the code structure
+            // and tests demand this for now.
+            for (let groupTypeIndex = 0; groupTypeIndex < this.db.MAX_GROUP_TYPES_PER_TEAM; ++groupTypeIndex) {
+                const key = `$group_${groupTypeIndex}`
+                delete properties[key]
+            }
+        }
```

Comment:
Just to confirm my understanding: we're removing group properties from the event here to match what we do for `$set` and friends, correct? 
But we keep group info, then we'll want to have these here.
I'd propose that we move everything groups into a separate step and handle it all there and not do anything within `normalizeEventStep` - because later we might want to make that also look at their add-on selection - i.e. skip group properties if they have disabled that add-on

Reply:
This is a separate follow-up I think we should do, sorry for the confusion here. 
What I'm thinking is 
1. we move all group stuff into a separate step
2. we make that step be ignored if `$process_person=false` or group analytics addon is disabled